soh/soh/Enhancements/TimeDisplay/TimeDisplay.cpp

```diff
@@ -0,0 +1,255 @@
+#define COLOR_LIGHT_BLUE ImVec4(0, 0.88f, 1.0f, 1.0f)
+#define COLOR_LIGHT_GREEN ImVec4(0.52f, 1.0f, 0.23f, 1.0f)
+#define COLOR_GREY ImVec4(0.78f, 0.78f, 0.78f, 1.0f)
+
+std::vector<std::pair<std::string, const char*>> digitList = {
+    { "DIGIT_0_TEXTURE", gCounterDigit0Tex },
+    { "DIGIT_1_TEXTURE", gCounterDigit1Tex },
+    { "DIGIT_2_TEXTURE", gCounterDigit2Tex },
+    { "DIGIT_3_TEXTURE", gCounterDigit3Tex },
+    { "DIGIT_4_TEXTURE", gCounterDigit4Tex },
+    { "DIGIT_5_TEXTURE", gCounterDigit5Tex },
+    { "DIGIT_6_TEXTURE", gCounterDigit6Tex },
+    { "DIGIT_7_TEXTURE", gCounterDigit7Tex },
+    { "DIGIT_8_TEXTURE", gCounterDigit8Tex },
+    { "DIGIT_9_TEXTURE", gCounterDigit9Tex },
+    { "COLON_TEXTURE",   gCounterColonTex },
+};
+
+std::vector<TimeObject> timeDisplayList = {
+    { DISPLAY_IN_GAME_TIMER,        "Display Gameplay Timer",       CVAR_ENHANCEMENT("TimeDisplay.Timers.InGameTimer") },
+    { DISPLAY_TIME_OF_DAY,          "Display Time of Day",          CVAR_ENHANCEMENT("TimeDisplay.Timers.TimeofDay") },
+    { DISPLAY_CONDITIONAL_TIMER,    "Display Conditional Timer",    CVAR_ENHANCEMENT("TimeDisplay.Timers.HotWater") },
+    { DISPLAY_NAVI_TIMER,           "Display Navi Timer",           CVAR_ENHANCEMENT("TimeDisplay.Timers.NaviTimer") }
+};
+
+std::vector<TimeObject> activeTimers;
+
+std::string convertDayTime(uint32_t dayTime) {
+    uint32_t totalSeconds = 24 * 60 * 60;
+    uint32_t ss = static_cast<uint32_t>(static_cast<double>(dayTime) * (totalSeconds - 1) / 65535);
+    uint32_t hh = ss / 3600;
+    uint32_t mm = (ss % 3600) / 60;
+    return fmt::format("{:0>2}:{:0>2}", hh, mm);
+}
+
+std::string convertNaviTime(uint32_t value) {
+    uint32_t totalSeconds = value * 0.05;
+    uint32_t ss = totalSeconds % 60; 
+    uint32_t mm = totalSeconds / 60;
+    return fmt::format("{:0>2}:{:0>2}", mm, ss);
+}
+
+std::string formatHotWaterDisplay(uint32_t value) {
+    uint32_t ss = value % 60;
+    uint32_t mm = value / 60;
+    return fmt::format("{:0>2}:{:0>2}", mm, ss);
+}
+
+std::string formatTimeDisplay(uint32_t value) {
+    uint32_t sec = value / 10;
+    uint32_t hh = sec / 3600;
+    uint32_t mm = (sec - hh * 3600) / 60;
+    uint32_t ss = sec - hh * 3600 - mm * 60;
+    uint32_t ds = value % 10;
+    return fmt::format("{}:{:0>2}:{:0>2}.{}", hh, mm, ss, ds);
+}
+
+void TimeDisplayGetTimer(uint32_t timeID) {
+    timeDisplayTime = "";
+    textureDisplay = 0;
+    textColor = COLOR_WHITE;
+
+    Player* player = GET_PLAYER(gPlayState);
+    uint32_t timer1 = gSaveContext.timer1Value;
+
+    switch (timeID) {
+        case DISPLAY_IN_GAME_TIMER:
+            textureDisplay = Ship::Context::GetInstance()->GetWindow()->GetGui()->GetTextureByName("GAMEPLAY_TIMER");
+            timeDisplayTime = formatTimeDisplay(GAMEPLAYSTAT_TOTAL_TIME).c_str();
+            break;
+        case DISPLAY_TIME_OF_DAY:
+            if (gSaveContext.dayTime >= 17759 && !(gSaveContext.dayTime >= 49155)) {
```

Comment:
Could the CLOCK_TIME macro be used for these (I know in the codebase we haven't adopted it from decomp yet, but we still have the macro at least)

Reply:
It looks like, with CLOCK_TIME, you still need to feed it an **hr** and a **min** which needs to be determined using the same process I'm using for convertDayTime(), not sure this really helps at all or saves any time/energy?

Reply:
I meant instead of the 17759 and 49155 numbers, so that its clear what times these mean

Reply:
I agree, clarification of the times values those represent with the macro would be useful. Also maybe why those values were used, not sure if I'm following the use case.

Reply:
Would something like this work?
![image](https://github.com/user-attachments/assets/78838b67-90f6-493a-989c-bc1728db4fca)

for clarification:
0 - 600 (Navi will spawn at 600)
601 - 3000 (Navi is active)
3001 - 25800 (Navi is gone, the timer resets to 0 when it hits 25800 where Navi can then spawn once more at 600)

Reply:
That wouldn't be everything you'd add to the enum, right? I think I'd be OK with that for the actual values passed, but I'd still like a human-readable time of day *somewhere* for what those values represent.

Reply:
Especially since the 17559 and 49155 are actual time values rather than time ranges.

Reply:
Oops got confused with Navi Timer instead of Time of Day. Yea I can enum the Start of Day and Start of Night values.